### PR TITLE
Feature/hector.add token info

### PR DESCRIPTION
**Pull request by fstar1129**
1. Added a empty array/null check for market stats data.
2. Update useEffect for reduce ext API call

### CODE REVIEW

File: src/contexts/TokenData.js [hunk 1]
```diff
@@ -397,24 +397,24 @@ const getTokenData = async (address, ethPrice, ethPriceOld) => {
...
-    const currentLiquidityUSD = data?.totalLiquidity * ethPrice * data?.derivedETH
-    const oldLiquidityUSD = oneDayData?.totalLiquidity * ethPriceOld * oneDayData?.derivedETH
+    // const currentLiquidityUSD = data?.totalLiquidity * ethPrice * data?.derivedETH
+    // const oldLiquidityUSD = oneDayData?.totalLiquidity * ethPriceOld * oneDayData?.derivedETH
 
     // set data
-    data.priceUSD = data?.derivedETH * ethPrice
-    data.totalLiquidityUSD = currentLiquidityUSD
+    // data.priceUSD = data?.derivedETH * ethPrice
+    // data.totalLiquidityUSD = currentLiquidityUSD
```
Comment (SarjuHansaliya) [inline src/contexts/TokenData.js]: @fstar1129 I see you have commented all these but we are using this in `src/pages/TokenPage.js` so we can't remove those field here.

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]

**Comment by fstar1129** [comment]
@SarjuHansaliya , I updated this PR for reducing coingecko call when getting initial data.
It should solve multi call of coingecko when data won't change.

**Comment by SarjuHansaliya** [comment]
@fstar1129 I believe there will be some issue in this implementation. for e.g. 2nd useEffect is dependant on `ethPrice, ethPriceOld` so whenever those value changes, it will call `getTokenData` but getTokenData data doesn't have any coingecko related data so final tokenData will not have those data.

I think we can make this more better by creating a memoized hook whose sole responsibility is to get Coin data from CoinGecko. that you can use in src/pages/TokenPage.js separately. 

what do you think @bmino ?

**Comment by SarjuHansaliya** [comment]
I have handled this in #62 , so closing this now.

**Pull request closed without merging**